### DISCUSSION TITLE
Show notification and force if 2FA is required when first login

### DISCUSSION THREAD

**Discussion by tapashdatta** [dani-garcia/vaultwarden#2604] [Ideas]
This seems to be a common issue for people where organisation sets 2FA is mandatory. But the user doesn’t receive a prompt to set up 2FA when they first login. It’s a bit confusing without knowing where the options are located. So, it would be better to have a prompt  forcing  to setup 2FA before proceeding further and give a direct link to the 2FA setup page. Thank for the app. Great effort!!

**Comment by BlackDex** [comment]
I'm afraid that this kind of logic needs to be build-in into the clients, and is not something which can be done on the server side.
On the server side we can't actually force someone to go to a specific page within the whole web-vault interface.

**Comment by tapashdatta** [comment]
Thanks for your reply. Just to add,  this is not an issue if 2FA is not forced.

**Comment by realMrTom** [comment]
But you may customize at least the mails the server sends.

If you set 2FA as mandatory and invite a user:
* the user gets an invitation message
* and a welcome message
both didn't tell the user he has to setup 2FA. So the user thinks he's done.

At the admin side the invited user stays in state "invited" (until the user has setup 2FA) and the admin
* cannot finally accept the invitation
* didn't know if the user hasn't received the mail, is not interested, forgot to click on the invitation link, hasn't setup 2FA, ...
